services/payments/statemachine_solana.go

```diff
@@ -352,17 +369,21 @@ func (sm *SolanaMachine) makeInstructions(
 	}, nil
 }
 
-func checkStatus(ctx context.Context, signature string, client *solanaClient.Client) (TxnCommitmentStatus, error) {
+func checkStatus(
+	ctx context.Context,
+	signature string,
+	client *solanaClient.Client,
+) (TxnCommitmentStatus, error) {
 	sigStatus, err := client.GetSignatureStatus(ctx, signature)
 	if err != nil {
 		return TxnUnknown, fmt.Errorf("status check error: %w", err)
 	}
 
 	if sigStatus == nil {
-		return TxnNotFound, nil
+		return TxnNotFound, &SolanaTransactionNotFoundError{}
```

Comment:
Again, instead of this you'd have returned a sentinel error, and then would have directly compared by `errors.Is` which is a bit cleaner for this use case than `errors.As`. The latter is more helpful in cases where error types do have some data and state in them, but here it seems like the struct error approach is either a premature decision, or just not a good tool for the job.